cli/src/commands/developer/scan.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::CurrentNetwork;
+
+use snarkvm::prelude::{Block, Plaintext, Record, ViewKey};
+
+use anyhow::{bail, Result};
+use clap::Parser;
+use std::str::FromStr;
+
+// TODO (raychu86): Figure out what to do with this naive scan. This scan currently does not check if records are already spent.
+/// Scan the snarkOS node for records.
+#[derive(Debug, Parser)]
+pub struct Scan {
```

Comment:
Most unspent records will likely be more towards the newest blocks than the oldest ones, perhaps we can put in options for scanning for a max balance (for example, just enough to make a transfer) and scanning in reverse.

Here are some thoughts in code about that: https://github.com/AleoHQ/snarkOS/pull/2237/files